src/Controller.js

```diff
@@ -0,0 +1,157 @@
+(function exportController() {
+  class Controller {
+    constructor(ship) {
+      this.ship = ship;
+
+      this.initialiseSea();
+
+      this.headsUpMessageBoard();
+
+      const sailButton = document.querySelector("#sailbutton");
+      sailButton.addEventListener("click", () => {
+        if (document.querySelector("#sailbutton").disabled) {
+          console.log("button disabled");
+        } else {
+          this.setSail();
+        }
+      });
+    }
+    get messageCurrentPort() {
+      return this.ship.currentPort.name;
+    }
+
+    get messageNextPort() {
+      const ship = this.ship;
+      const indexOfNextPort =
+        ship.itinerary.ports.indexOf(ship.currentPort) + 1;
+      let nextPort;
+      if (!this.endOfItinerary) {
+        nextPort = ship.itinerary.ports[indexOfNextPort].name;
+      } else {
+        nextPort = "End of cruise";
+      }
+      return nextPort;
+    }
+
+    get endOfItinerary() {
+      const indexOfNextPort =
+        this.ship.itinerary.ports.indexOf(this.ship.currentPort) + 1;
+      return indexOfNextPort >= this.ship.itinerary.ports.length;
+    }
+
+    initialiseSea() {
+      const backgrounds = ["../images/water0.png", "../images/water1.png"];
```

Comment:
The animation wasnt working when I tried to load this, the file path here is a bit off. Should stay in the same directory instead of jumping up one:
```
      const backgrounds = ["./images/water0.png", "./images/water1.png"];
```